src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -76,15 +24,15 @@ public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder co
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            var projectPath = configuration.GetProjectPath();
-            if (string.IsNullOrEmpty(projectPath))
+            var fileProvider = configuration.GetFileProvider();
+            if (fileProvider == null)
             {
                 throw new InvalidOperationException(Resources.FormatError_MissingBasePath(
```

Comment:
Nit: should probably update the resource txt/name to refer to FileProvider now instead if we are keeping this check.